.github/workflows/ci.yml

```
@@ -0,0 +1,80 @@
+on: push
+
+jobs:
+  code-quality:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: erlef/setup-beam@v1
+        with:
+          # Just some default versions to run code quality tools with.
+          # These can be upgraded as and when but probably won't need to change much.
+          otp-version: "24.0.6"
+          elixir-version: "1.12.3"
+      - uses: actions/cache@v2
+        with:
+          path: |
+            deps
+            _build
+            priv/plts
+          key: ${{ runner.os }}-mix-${{ hashFiles('**/mix.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-mix-
+      - run: mix deps.get
+      - run: mix credo --strict
+        continue-on-error: true
+      - run: mix format --check-formatted
+        continue-on-error: true
+      - run: mix dialyzer
+  test:
+    strategy:
+      matrix:
+        os: 
+          - ubuntu-latest
+        otp-version:
+          - "24.0.6"
+        elixir-version: 
+          - "1.12.3"
+        mongo-version:
+          - maj: 4
+            min: 4
+            patch: 8
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: erlef/setup-beam@v1
+        with:
+          otp-version: ${{ matrix.otp-version }}
+          elixir-version: ${{ matrix.elixir-version }}
+      - uses: actions/cache@v2
+        with:
+          path: |
+            deps
+            _build
+          key: ${{ runner.os }}-mix-${{ hashFiles('**/mix.lock') }}
```

If there is a proper matrix of Elixir and OTP versions, you will want to include the versions in the cache key in order to pull the proper cache version.

---

Yes, more important here.  I'll fix.

---

```suggestion
          key: ${{ runner.os }}-{{runner.otp-version}}-{{runner.elixir-version}}-mix-${{ hashFiles('**/mix.lock') }}
```

I think this is what @ankhers  is suggesting!